functions/index.js

```diff
@@ -225,3 +225,20 @@ exports.regionSubscribeCreate = functions.region('europe-west1').firestore.docum
       console.log('ID', snap.id);
     }
   });
+
+  exports.reportedPostsCreate = functions.region('europe-west1').firestore.document('/reported-posts/{reportRequestId}')
+  .onCreate(async (snap, context) => {
+    try {
+      const db = admin.firestore();
+      const snapValue = snap.val();
```

Comment:
should be snap.data()

Reply:
I could only find `.val()` in the documentation, assuming that I was looking at the right [snapshot](https://firebase.google.com/docs/reference/functions/providers_database_.datasnapshot#methods) in the [onCreate](https://firebase.google.com/docs/reference/functions/providers_database_.refbuilder#on-create) function

Reply:
turns out snap.data() was correct :D